runtime/src/iree/hal/cts/semaphore_submission_test.h

```diff
@@ -544,9 +544,84 @@ TEST_P(semaphore_submission_test, IntermediateSemaphoreBetweenDeviceBatches) {
   iree_hal_semaphore_release(semaphore2);
 }
 
-// TODO: test device -> device synchronization: submit multiple batches with
+// Test device -> device synchronization: submit multiple batches with
 // multiple later batches waiting on the same signaling from a former batch.
-//
+TEST_P(semaphore_submission_test, TwoBatchesWaitingOn1FormerBatchAmongst2) {
```

Comment:
This may be timing out on Metal: https://github.com/openxla/iree/actions/runs/8730836551/job/23955300868#step:7:3016

Reply:
Thank you for pointing this out. I did not notice.
It seems that it is failing like that pretty consistently.

https://github.com/openxla/iree/actions/runs/8739216455/job/23980063767#step:7:2996
https://github.com/openxla/iree/actions/runs/8732772803/job/23960330556#step:7:2996
https://github.com/openxla/iree/actions/runs/8732914924/job/23960661155#step:7:2996

Maybe I can add an expected failure for Metal and fix it later.

Reply:
You can exclude the entire `semaphore_submission` test from the Metal HAL CTS here: https://github.com/openxla/iree/blob/244195960ff1ad60147c351b0c91155d0c9cd7fb/runtime/src/iree/hal/drivers/metal/cts/CMakeLists.txt#L20-L22

OR disable this individual test case across all backends by renaming from `TEST_P(semaphore_submission_test, TwoBatchesWaitingOn1FormerBatchAmongst2)` to `TEST_P(semaphore_submission_test, DISABLED_TwoBatchesWaitingOn1FormerBatchAmongst2)`

There isn't currently a nice way to disable a single test case for only one backend.

Reply:
Curiously, on my machine with the HIP runtime if I run
```
IntermediateSemaphoreBetweenDeviceBatches
TwoBatchesWaitingOn1FormerBatchAmongst2
```
in that order, the second test hangs at the test setup stage.

If I run it on its own it is fine.

Reply:
@sogartar are you working on a PR to exclude the failing tests in some way or disable them? We may just want to revert this... don't want CI test failures on tip of tree for this long.

Reply:
I've put up #17100 to disable metal side for now.